src/server/scripts/OutdoorPvP/OutdoorPvPNA.cpp

```diff
@@ -854,7 +824,59 @@ class OutdoorPvP_nagrand : public OutdoorPvPScript
     }
 };
 
+class OutdoorPvP_nagrandCreature : public CreatureScript
+{
+public:
+    OutdoorPvP_nagrandCreature() : CreatureScript("OutdoorPvP_nagrandCreature") { }
+
+    CreatureAI* GetAI(Creature* creature) const override
+    {
+        return new npc_halaa_creature(creature);
+    }
+
+    struct npc_halaa_creature : public ScriptedAI
+    {
+
+        bool m_initialized = false;
+        npc_halaa_creature(Creature* creature) : ScriptedAI(creature) { }
+
+        void UpdateAI(uint32 diff) override
+        {
+            if (!m_initialized)
+            {
+                std::list<Creature*> creatures;
+                uint32 entry = 0;
+                for (int i = 0; i < 12; i++)
+                {
+                    me->GetCreatureListWithEntryInGrid(creatures, PatrolCreatureEntry[i].idPatrol, 250);
+                }
+
+                if (creatures.size() == 40)
+                {
+                    m_initialized = true;
+                    for (std::list<Creature*>::iterator itr = creatures.begin(); itr != creatures.end(); ++itr)
+                    {
+                        Creature* const creature = *itr;
+                        if (entry < 20)
+                        {
+                            halaaNPCHorde[entry] = (creature->GetSpawnId());
+                        }
+                        else
+                        {
+                            halaaNPCAlly[entry-20] = (creature->GetSpawnId());
+                        }
+                        creature->AddObjectToRemoveList();
+                        entry++;
+                        sObjectMgr->RemoveCreatureFromGrid(creature->GetSpawnId(), creature->GetCreatureData());
+                    }
+                }
+            }
+        }
+    };
+};
+
 void AddSC_outdoorpvp_na()
 {
     new OutdoorPvP_nagrand();
+    new OutdoorPvP_nagrandCreature();
```

Comment:
Use new creature script instead. RegisterCreatureScript()

Reply:
What do you mean with that Kitzunu? I have to use RegisterCreatureScript(name) or? because i updated it thinking thats what you mean but AI now doesnt work so i think that isn't what you mean :/, and i was looking for RegisterCreatureScript but i only found those: 
![image](https://user-images.githubusercontent.com/37176834/229488298-f12f9a19-7628-4833-9af0-bbac41ff4d93.png)

I don't know then how to do, im looking someothers creaturescript